From 01ab34d927143aad8e2617ac3f9bfa172d8a0d0f Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Tue, 1 Aug 2017 14:40:24 +0800
Subject: [PATCH] ARM: dts: rockchip: add usb ohci node for rk3288

rk3288 and rk3288w have an ohci controller, although rk3288
ohci doesn't actually work on hardware, but rk3288w ohci can
work well, so we need to add usb ohci node in dts, and then
add a quirk in ohci driver to disable ohci for rk3288.

Change-Id: I0ce26a9851f6bc3c82cc8b7dc16ecc7b4ce2ca28
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 arch/arm/boot/dts/rk3288.dtsi | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
index 01cbfff619e0..d0f04f43b815 100644
--- a/arch/arm/boot/dts/rk3288.dtsi
+++ b/arch/arm/boot/dts/rk3288.dtsi
@@ -612,7 +612,20 @@
 		status = "disabled";
 	};
 
-	/* NOTE: ohci@ff520000 doesn't actually work on hardware */
+	/*
+	 * NOTE: ohci@ff520000 doesn't actually work on rk3288
+	 * hardware, but can work on rk3288w hardware.
+	 */
+	usb_host0_ohci: usb@ff520000 {
+		compatible = "generic-ohci";
+		reg = <0x0 0xff520000 0x0 0x20000>;
+		interrupts = <GIC_SPI 41 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&cru HCLK_USBHOST0>;
+		clock-names = "usbhost";
+		phys = <&usbphy1>;
+		phy-names = "usb";
+		status = "disabled";
+	};
 
 	usb_host1: usb@ff540000 {
 		compatible = "rockchip,rk3288-usb", "rockchip,rk3066-usb",
-- 
2.35.3

